components/org.wso2.carbon.identity.query.saml/src/main/java/org/wso2/carbon/identity/query/saml/SignKeyDataHolder.java

```diff
@@ -0,0 +1,242 @@
+        String keyAlias;
+        KeyStoreAdmin keyAdmin;
+        KeyStoreManager keyMan;
+        Certificate[] certificates;
+        int tenantID;
+
+        try {
+
+            if (tenantDomain == null) {
+                tenantDomain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+            }
+
+            tenantID = SAMLSSOUtil.getRealmService().getTenantManager().getTenantId(tenantDomain);
+
+            IdentityTenantUtil.initializeRegistry(tenantID, tenantDomain);
+
+            if (tenantID != MultitenantConstants.SUPER_TENANT_ID) {
+                String keyStoreName = SAMLSSOUtil.generateKSNameFromDomainName(tenantDomain);
+                keyAlias = tenantDomain;
+                keyMan = KeyStoreManager.getInstance(tenantID);
+                KeyStore keyStore = keyMan.getKeyStore(keyStoreName);
+                issuerPK = (PrivateKey) keyMan.getPrivateKey(keyStoreName, tenantDomain);
+                certificates = keyStore.getCertificateChain(keyAlias);
+                issuerCerts = new X509Certificate[certificates.length];
+
+                int i = 0;
+                for (Certificate certificate : certificates) {
+                    issuerCerts[i++] = (X509Certificate) certificate;
+                }
+
+                signatureAlgorithm = XMLSignature.ALGO_ID_SIGNATURE_RSA;
+
+                publicKey = issuerCerts[0].getPublicKey();
+                String pubKeyAlgo = publicKey.getAlgorithm();
+                if (DSA_ENCRYPTION_ALGORITHM.equalsIgnoreCase(pubKeyAlgo)) {
+                    signatureAlgorithm = XMLSignature.ALGO_ID_SIGNATURE_DSA;
+                }
+
+            } else {
+                keyAlias = ServerConfiguration.getInstance().getFirstProperty(
+                        SECURITY_KEY_STORE_KEY_ALIAS);
+
+                keyAdmin = new KeyStoreAdmin(tenantID,
+                        SAMLSSOUtil.getRegistryService().getGovernanceSystemRegistry());
+                keyMan = KeyStoreManager.getInstance(tenantID);
+
+                issuerPK = (PrivateKey) keyAdmin.getPrivateKey(keyAlias, true);
+
+                certificates = keyMan.getPrimaryKeyStore().getCertificateChain(keyAlias);
+
+                issuerCerts = new X509Certificate[certificates.length];
+
+                int i = 0;
+                for (Certificate certificate : certificates) {
+                    issuerCerts[i++] = (X509Certificate) certificate;
+                }
+
+                signatureAlgorithm = XMLSignature.ALGO_ID_SIGNATURE_RSA;
+
+                publicKey = issuerCerts[0].getPublicKey();
+                String pubKeyAlgo = publicKey.getAlgorithm();
+                if (DSA_ENCRYPTION_ALGORITHM.equalsIgnoreCase(pubKeyAlgo)) {
+                    signatureAlgorithm = XMLSignature.ALGO_ID_SIGNATURE_DSA;
+                }
+            }
+
+        } catch (IdentityException e) {
+            log.error(e);
+
+        } catch (Exception e) {
```

Comment:
Any particular reason why you are catching Exception class instead of specific classes?